csm/core/controllers/rgw/s3/bucket.py

```diff
@@ -0,0 +1,98 @@
+from csm.core.controllers.validators import ValidationErrorFormatter
+from csm.core.controllers.rgw.s3.base import S3BaseView, S3BaseSchema
+
+class LinkBucketSchema(S3BaseSchema):
+
+    """
+    S3 Bucket Link schema validation class.
+    """
+    uid = fields.Str(data_key=const.RGW_JSON_UID, required=True)
+    bucket = fields.Str(data_key=const.RGW_JSON_BUCKET, required=True)
+    bucket_id = fields.Str(data_key=const.RGW_JSON_BUCKET_ID, missing=None)
+
+class UnlinkBucketSchema(S3BaseSchema):
+
+    """
+    S3 Bucket Unlink schema validation class.
+    """
+    uid = fields.Str(data_key=const.RGW_JSON_UID, required=True)
+    bucket = fields.Str(data_key=const.RGW_JSON_BUCKET, required=True)
+
+class schemaFactory:
+    operation_schema_map = {
+        "link": LinkBucketSchema,
+        "unlink": UnlinkBucketSchema
+    }
+    @staticmethod
+    def init(operation):
+        return operation_schema_map[operation]()
+
+@CsmView._app_routes.view("/api/v2/s3/bucket")
+class S3BucketView(S3BaseView):
+
+    """
+    S3 Bucket View for REST API implementation.
+    PUT: Bucket Opertation
+    """
+
+    def __init__(self, request):
+        """S3 Bucket View Init."""
+        super().__init__(request, const.RGW_S3_BUCKET_SERVICE)
+
+    @CsmAuth.permissions({Resource.S3_BUCKET: {Action.UPDATE}})
+    @Log.trace_method(Log.DEBUG)
+    async def put(self):
+        """
+        PUT REST implementation for  s3 bucket.
+        """
+        Log.debug(f"Handling s3 bucket PUT request"
+                  f" user_id: {self.request.session.credentials.user_id}")
+        # Load Expected Bucket Schema
+        operation = None
+        operation_arguments = None
+        try:
+            # Validate Request Schema Body with BucketSchema
+            input_body = await self.request.json()
+            if "operation" not in input_body.keys() or "arguments" not in input_body.keys():
+                 raise ValidationError("operation and arguments are required fields")
+            operation = input_body['operation']
+            if operation not in schemaFactory.operation_schema_map:
+                raise ValidationError(f"{operation}: is not supported")
+            operation_arguments = input_body['arguments']
+            if not isinstance(operation_arguments, dict):
+                raise ValidationError(f" operation_arguments is not of Type Dict :- {operation_arguments}")
+            # Validation Operation Level Schema Validation
+            schema = schemaFactory.init(operation)
+            request_body = schema.load(operation_arguments)
+            Log.debug(f"Handling s3 bucket PUT request"
+                  f" request body: {request_body}")
+        except json.decoder.JSONDecodeError:
+            raise InvalidRequest(message_args="Invalid Request Body")
+        except ValidationError as val_err:
+            raise InvalidRequest(f"{ValidationErrorFormatter.format(val_err)}")
```

Comment:
Submitted one patch for validation logic. Kindly check - https://github.com/abhijit1patil/cortx-manager/pull/2